### ISSUE TITLE
Rename MapboxDirections.swift to Mapbox Directions for Swift

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-directions-swift#399]
This project was originally named MapboxDirections.swift because it started as a single file named MapboxDirections.swift in 6f7300441796fce9e70ba4a31f532800d1c09ba2. (Lore has it that, even before this repository existed, it was a gist by the same name.) Over the years, the “dot swift” extension has been a bit of a mouthful and even occasionally a source of build issues: https://github.com/mapbox/MapboxDirections.swift/issues/318#issuecomment-464834925. Having “.swift” in the name also caused confusion back when the library also supported Objective-C. We’ve just dropped Objective-C support, but emphasizing Swift in contrast to Objective-C matters less and less these days in Apple-centric environments like CocoaPods.

For v1.0, we’re planning to rename the project to “Mapbox Directions for Swift”, the repository to mapbox/mapbox-directions-swift, and the pod to simply “MapboxDirections”. Any developer installing the library using CocoaPods would need to replace `MapboxDirections.swift` with `MapboxDirections` in their Podfile. Otherwise, there’s no immediate impact to developers, since GitHub automatically redirects renamed repositories.

The steps to rename look roughly like this:

* [x] Replace occurrences of “MapboxDirections.swift” in the repository: #400
* [x] Rename the repository to mapbox/mapbox-directions-swift
* [x] Update Bitrise configuration
* [x] Update CircleCI configuration
* [x] Update Codecov configuration
* [x] Publish v1.0 (and any prereleases) to a new “MapboxDirections” pod: #398
* [x] Deprecate the MapboxDirections.swift pod in favor of the MapboxDirections pod
* [x] Update documentation across docs.mapbox.com

CocoaPods doesn’t have a built-in way to rename a pod, so we’ll have to create a new pod and deprecate the old one. We could continue to publish the pod under both names for a while, but I’m not sure that maintenance overhead is really necessary given the disruption that v1.0 will cause regardless.

When we created the Mapbox Speech for Swift project, we avoid the “.swift” nomenclature in anticipation of a similar rename across all the other Swift service libraries, including MapboxGeocoder.swift and MapboxStatic.swift. That’ll be tracked in separate issues in the respective repositories.

(This rename was originally attempted in #366 and #362.)

/cc @mapbox/navigation-ios @mapbox/docs @mapbox/ios @mapbox/driver-app

**Comment by 1ec5** [comment]
> Deprecate the MapboxDirections.swift pod in favor of the MapboxDirections pod

The command didn’t go as I had expected, but [MapboxDirections.swift does appear to be deprecated](https://cocoapods.org/pods/MapboxDirections.swift/) in favor of [MapboxDirections](https://cocoapods.org/pods/MapboxDirections/):

```
$ pod trunk deprecate 'MapboxDirections.swift' --in-favor-of=MapboxDirections
[!] An unexpected error occurred: <!DOCTYPE html>
	<html>
	  <head>
		<meta name="viewport" content="width=device-width, initial-scale=1">
		<meta charset="utf-8">
		<title>Application Error</title>
		<style media="screen">
		  html,body,iframe {
			margin: 0;
			padding: 0;
		  }
		  html,body {
			height: 100%;
			overflow: hidden;
		  }
		  iframe {
			width: 100%;
			height: 100%;
			border: 0;
		  }
		</style>
	  </head>
	  <body>
		<iframe src="//www.herokucdn.com/error-pages/application-error.html"></iframe>
	  </body>
	</html>
```

**Comment by 1ec5** [comment]
Fixed by CocoaPods/Specs@1baaee87df90aeb344fa3236fe3d74da1865a516.
